### PR TITLE
FIX #806 Part 1

### LINKED ISSUE #806: Remove Apache Commons Lang dependency
Following with our effort to reduce dependencies, the Apache Commons Lang doesn't add much value and thus must be removed

To remove from `build.gradle`: `org.apache.commons:commons-lang3:3.5`

### PR DESCRIPTION
**Pull request by alexjavabraz**
Fix #806 

- removed dependency commons.ListUtil methods
- new methods with similar 'commons lib' behavior
- created some tests 

Pending

- StringUtils methods
- Pair

### CODE REVIEW

**Comment by tinchou** [comment]
Hi and thanks for your contribution! It looks great overall, but let me make some suggestions.

With regards to CircleCI, we have a known issue that prevents our integration process to run for external contributors.

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        Arrays.setAll(byteObjectArray, n -> primitiveByteArray[n]);
+
+        return byteObjectArray;
+    }
+
+    /**
+     * Checks if an array of Object is not empty and not null.
+     * @return
+     */
+    public static boolean isEmpty(final Object[] o){
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: This overload is never used in production code, so it should be removed

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        return (o == null || o.length == 0);
+    }
+
+    /**
+     * Checks if an array of Object is not empty and not null.
+     * @param Object[] o
+     * @return
+     *     true if the array is not empty and not null
+     */
+    public static boolean isNotEmpty(final Object[] o){
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: This overload is never used in production code, so it should be removed

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        return !(o == null || o.length == 0);
+    }
+
+    /**
+     * Checks if an array of primitive bytes is not empty and not null.
+     * @param array
+     * @return
+     *     true if the array is not empty and not null
+     */
+    public static boolean isNotEmpty(byte[] array){
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: This overload is not being tested

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    public static boolean isEmpty(final Object[] o){
+        return (o == null || o.length == 0);
+    }
+
+    /**
+     *  Checks if an array of Object is not empty and not null.
+     * @param o
+     * @return
+     */
+    public static boolean isEmpty(final byte[] o){
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: This overload is not being tested

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    public static List<Byte> asByteList(final byte[] primitiveByteArray) {
+        Byte[] arrayObj = convertTo(primitiveByteArray);
+        return Arrays.asList(arrayObj);
+    }
+
+    /**
+     * @param primitiveByteArray
+     * @return
+     */
+    public static Byte[] convertTo(@Nonnull final byte[] primitiveByteArray){
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: Make this `private` until you need to make it `public` for an actual feature.

File: rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public static boolean isNotEmpty(byte[] array){
+        return !(array == null || array.length == 0);
+    }
+
+    public static int getLength(byte [] array){
+        if (array == null) {
+            return 0;
+        }
+        return Array.getLength(array);
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/util/ListArrayUtil.java]: Here you're using the `java.lang.reflect.Array` class, which is unnecessary.

```suggestion
        return array.length;
```

File: rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java [hunk 7]
```diff
@@ -19,23 +19,21 @@
...
-
-import java.math.BigInteger;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
-import static org.apache.commons.lang3.ArrayUtils.nullToEmpty;
-import static org.ethereum.util.ByteUtil.EMPTY_BYTE_ARRAY;
+    import co.rsk.util.ListArrayUtil;
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: Please indent the imports correctly

File: rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java [hunk 8]
```diff
@@ -75,13 +73,13 @@ public TransactionReceipt(byte[] rlp) {
         RLPList logs = (RLPList) receipt.get(3);
         RLPItem gasUsedRLP = (RLPItem) receipt.get(4);
 
-        postTxState = nullToEmpty(postTxStateRLP.getRLPData());
+        postTxState = ListArrayUtil.nullToEmpty(postTxStateRLP.getRLPData());
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: Add a static import so the diff is smaller :)

File: rskj-core/src/main/java/org/ethereum/vm/program/InternalTransaction.java [hunk 9]
```diff
@@ -38,7 +37,7 @@
     public InternalTransaction(byte[] parentHash, int deep, int index, byte[] nonce, DataWord gasPrice, DataWord gasLimit,
                                byte[] sendAddress, byte[] receiveAddress, byte[] value, byte[] data, String note) {
 
-        super(nonce, getData(gasPrice), getData(gasLimit), receiveAddress, nullToEmpty(value), nullToEmpty(data));
+        super(nonce, getData(gasPrice), getData(gasLimit), receiveAddress, ListArrayUtil.nullToEmpty(value), ListArrayUtil.nullToEmpty(data));
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/vm/program/InternalTransaction.java]: Add static imports to make the diff smaller

File: rskj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 10]
```diff
@@ -211,7 +211,7 @@ public Program(
 
         this.invoke = programInvoke;
 
-        this.ops = nullToEmpty(ops);
+        this.ops = ListArrayUtil.nullToEmpty(ops);
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/vm/program/Program.java]: static imports

File: rskj-core/src/test/java/co/rsk/util/ListArrayUtilTest.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
+package co.rsk.util;
```
Comment (tinchou) [inline rskj-core/src/test/java/co/rsk/util/ListArrayUtilTest.java]: Please add the license header

**Comment by tinchou** [comment]
Hi Alex!

We've updated our infrastructure and you should now be able to run PR checks. This way it'll make it much easier for us to merge your changes, so let me know if we can work together on it! :)

**Comment by alexjavabraz** [comment]
> Hi Alex!
> 
> We've updated our infrastructure and you should now be able to run PR checks as an external contributor. This way it'll make it much easier for us to merge your changes, so let me know if we can work together on it! :)

Hi @collheesi , sure i'm in! Tks

**Comment by alexjavabraz** [comment]
@collheesi i'm reviewing all comments. Tks

**Comment by alexjavabraz** [comment]
@collheesi I'm closing this pr and open new one with you comments applied.

**Comment by collheesi** [comment]
Sure, no problem!

**Pull request closed without merging**